cellmlmanip/units.py

```diff
@@ -322,13 +424,22 @@ def traverse(self, expr: sympy.Expr):
                     # otherwise, keep the symbol
                     out = self.ureg.Quantity(expr, metadata.units)
             return out
-        elif expr.is_Integer:
-            out = int(expr) * self.ureg.dimensionless
-            return out
-        elif expr.is_Rational:
-            # NOTE: can't send back Rational(1,2) * u.dimensionless
-            # Used by Sympy for root e.g. sqrt(x) == Pow(x, Rational(1,2))
-            out = float(expr) * self.ureg.dimensionless
+
+        elif expr.is_Number:
+            units = self.ureg.dimensionless
+            for key in self.dummy_metadata:
```

Comment:
the dummy_metadata is the symbol dictionary created by reading in a model. So it has entries
{key_symbol: MetaDummyObject} for each symbol but it is the number field of the MetaDummy object that actually matches the expression. The key is whatever symbol the reader decided to use. ie. it might replace 'a' in cellml with 'symbol_a'.

So I need to loop thru each entry in the symbol dict and then see if it is the matching number in order to get the units.

e.g. the tests set up
        symbol_info = { (snipped)
            a: MetaDummy('a', ureg.meter, a),
           _1: MetaDummy('_1', ureg.kelvin, _1, number=sp.Float(1.0)),
        }

If you can suggest a better loop please do.

Reply:
OK. Can we change it to `for key, metadata in self.dummy_metadata.items()` at least? I'll make an issue for the weird keys